tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/ConcreteFunction.java

```diff
@@ -241,31 +241,31 @@ public String toString() {
     }
 
     List<Output<?>> outputList =
-        PartitionedCall.create(
-                scope,
-                inputList,
-                Arrays.stream(outputDtypes)
-                    .map(x -> TensorTypeRegistry.find(x).type())
-                    .collect(Collectors.toList()),
-                this)
-            .output();
-
-    Map<String, Operand<?>> namedOutputs = new LinkedHashMap<>(signature().outputNames().size());
-
-    List<String> outputNames = new ArrayList<>(signature().outputNames());
-    for (int i = 0; i < outputNames.size(); i++) {
-      String outputName = outputNames.get(i);
-
-      if (i > outputList.size()) {
-        throw new IllegalStateException(
-            "Somehow, not all required outputs were returned from the function");
+        PartitionedCall.create(scope, inputList, outputTypes, this).output();
+
+    if (signature().outputNames().size() == 0) {
+      return Collections.emptyMap();
+    } else if (signature().outputNames().size() == 1) {
+      return Collections.singletonMap(
+          signature().outputNames().iterator().next(), outputList.get(0));
+    } else {
+      Map<String, Operand<?>> namedOutputs = new LinkedHashMap<>(signature().outputNames().size());
+
+      Iterator<String> outputNames = signature().outputNames().iterator();
+      for (int i = 0; outputNames.hasNext(); i++) {
+        String outputName = outputNames.next();
+
+        if (i > outputList.size()) {
+          throw new IllegalStateException(
+              "Somehow, not all required outputs were returned from the function");
```

Comment:
Might be useful if this exception returned both `i` and `outputList.size()`. Though I suspect it's very unlikely to happen.

Reply:
Done. I've also move that sanity check outside the loop, it could be done beforehand.